### PR TITLE
[fix][broker] The configuration loadBalancerNamespaceMaximumBundles is invalid

### PR DESCRIPTION
**Pull request by lordcheng10**
### Motivation
The configuration loadBalancerNamespaceMaximumBundles is invalid.

example to illustrate:
1. Configure loadBalancerNamespaceMaximumBundles=128
1. The current number of bundles in the namespace is 120
2. According to the current logic, after executing the method findBundlesToSplit, 40 bundles may be found that need to be split;
3. After these bundles, the number of bundles in the namespace will become 120+40=160 > loadBalancerNamespaceMaximumBundles=128

modified logic：
1. When judging whether the number of bundles exceeds the configuration, it is necessary to add the already selected bundles：

image

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by Demogorgon314** [comment]
Can you add a unit test to cover this case?

**Comment by codelipenghui** [comment]
@lordcheng10 Could you please provide more details about the BUG? so that the reviewers can easily understand what the issue is that the PR wants to fix.

**Comment by lordcheng10** [comment]
> @lordcheng10 Could you please provide more details about the BUG? so that the reviewers can easily understand what the issue is that the PR wants to fix.

Sorry, I didn't describe clearly. @codelipenghui 

I'll give an example to illustrate:
1. Configure loadBalancerNamespaceMaximumBundles=128
1. The current number of bundles in the namespace is 120
2. According to the current logic, after executing the method findBundlesToSplit, 40 bundles may be found that need to be split;
3. After these bundles, the number of bundles in the namespace will become 120+40=160 > loadBalancerNamespaceMaximumBundles

**Comment by Technoboy-** [comment]
> > @lordcheng10 Could you please provide more details about the BUG? so that the reviewers can easily understand what the issue is that the PR wants to fix.
> 
> Sorry, I didn't describe clearly. @codelipenghui
> 
> I'll give an example to illustrate:
> 
> 1. Configure loadBalancerNamespaceMaximumBundles=128
> 2. The current number of bundles in the namespace is 120
> 3. According to the current logic, after executing the method findBundlesToSplit, 40 bundles may be found that need to be split;
> 4. After these bundles, the number of bundles in the namespace will become 120+40=160 > loadBalancerNamespaceMaximumBundles=128

Make sense.

**Comment by lordcheng10** [comment]
> > > @lordcheng10 Could you please provide more details about the BUG? so that the reviewers can easily understand what the issue is that the PR wants to fix.
> > 
> > 
> > Sorry, I didn't describe clearly. @codelipenghui
> > I'll give an example to illustrate:
> > 
> > 1. Configure loadBalancerNamespaceMaximumBundles=128
> > 2. The current number of bundles in the namespace is 120
> > 3. According to the current logic, after executing the method findBundlesToSplit, 40 bundles may be found that need to be split;
> > 4. After these bundles, the number of bundles in the namespace will become 120+40=160 > loadBalancerNamespaceMaximumBundles=128
> 
> Make sense.

When judging whether the number of bundles exceeds the configuration, it is necessary to add the already selected bundles. @codelipenghui @Technoboy-

**Review by Technoboy- — APPROVED** [review]

**Comment by lordcheng10** [comment]
> Can you add a unit test to cover this case?

added a unit test :  testLoadBalancerNamespaceMaximumBundles

PTAL,thanks! @Demogorgon314

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/BundleSplitterTaskTest.java [hunk 1]
```diff
@@ -95,6 +96,61 @@ public void testSplitTaskWhenTopicJustOne() {
...
+
+        try {
+            int currentBundleCount = pulsar.getNamespaceService().getBundleCount(NamespaceName.get("ten/ns"));
+            final Set<String> bundlesToSplit = bundleSplitterTask.findBundlesToSplit(loadData, pulsar);
+            Assert.assertEquals(bundlesToSplit.size() + currentBundleCount,
+                    pulsar.getConfiguration().getLoadBalancerNamespaceMaximumBundles());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```
Comment (Demogorgon314) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/BundleSplitterTaskTest.java:152]:
```suggestion
        int currentBundleCount = pulsar.getNamespaceService().getBundleCount(NamespaceName.get("ten/ns"));
        final Set<String> bundlesToSplit = bundleSplitterTask.findBundlesToSplit(loadData, pulsar);
        Assert.assertEquals(bundlesToSplit.size() + currentBundleCount,
                pulsar.getConfiguration().getLoadBalancerNamespaceMaximumBundles());
```

**Review by Demogorgon314 — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/BundleSplitterTaskTest.java (hunk 1 above)
Comment (lordcheng10) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/impl/BundleSplitterTaskTest.java]: fixed

**Comment by lordcheng10** [comment]
@codelipenghui @aloyszhang PTAL,thanks!

**Review by aloyszhang — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**